pyirf/interpolation.py

```diff
@@ -0,0 +1,81 @@
+"""Functions for performing interpolation of IRF to the values read from the data."""
+
+import numpy as np
+import astropy.units as u
+from scipy.interpolate import griddata
+
+
+def interpolate_effective_area_per_energy_and_fov(effective_area, grid_points, target_point, min_effective_area=1. * u.Unit('m2'), method='linear'):
+    """
+    Takes a grid of effective areas for a bunch of different parameters
+    and interpolates (log) effective areas to given value of those parameters
+
+    Parameters
+    ----------
+    effective_area: np.array of astropy.units.Quantity[area]
+        grid of effective area, of shape (n_grid_points, n_fov_offset_bins, n_energy_bins)
+    grid_points: np.array
+        list of parameters corresponding to effective_area, of shape (n_grid_points, n_interp_dim)
+    target_point: np.array
+        values of parameters for which the interpolation is performed, of shape (n_interp_dim)
+    min_effective_area: astropy.units.Quantity[area]
+        Minimum value of effective area to be considered for interpolation
```

Comment:
because the effective area changes by a few orders of magnitude the interpolation is much more efficient in log scale, but there are bins in which the effective area is zero, thus you cannot simply calculate the logarithm of it. So for the calculation for such empty bins log value of the min_effective area is used instead, and then after the interpolation is done everything that is less then 1.1 * effective area is set back to zero, therefore if you interpolate between two zeros  you get back zero, and if the interpolation is between zeros and some small values the algorithm still returns a reasonable value. Nevertheless the default value is set low enough (at the value that you normally have very poor statistics in the MCs) that such low effective areas will not have any effect in the analysis